test/contracts-network/colony-network-extensions.js

```diff
@@ -145,33 +155,83 @@ contract("Colony Network Extensions", (accounts) => {
       await metaColony.addExtensionToNetwork(TEST_EXTENSION, testExtension2Resolver.address);
     });
 
-    it("allows a root user to install an extension with any version", async () => {
-      await colony.installExtension(TEST_EXTENSION, 2, { from: ROOT });
+    it("allows a root user to install an extension", async () => {
+      const tx = await colony.installExtension(TEST_EXTENSION, 1, { from: ROOT });
 
-      const extensionAddress = await colonyNetwork.getExtensionInstallation(TEST_EXTENSION, colony.address);
-      const extension = await TestExtension2.at(extensionAddress);
+      const extensionAddress = getExtensionAddressFromTx(tx);
+      const extension = await TestExtension1.at(extensionAddress);
       const owner = await extension.owner();
       expect(owner).to.equal(colonyNetwork.address);
 
       const identifier = await extension.identifier();
       const version = await extension.version();
       const colonyAddress = await extension.getColony();
       expect(identifier).to.equal(TEST_EXTENSION);
-      expect(version).to.eq.BN(2);
+      expect(version).to.eq.BN(1);
       expect(colonyAddress).to.equal(colony.address);
 
       // Only colonyNetwork can install the extension
       await checkErrorRevert(extension.install(colony.address), "ds-auth-unauthorized");
     });
 
+    it("allows a root user to install an extension with any version", async () => {
+      const tx = await colony.installExtension(TEST_EXTENSION, 2, { from: ROOT });
+
+      const extensionAddress = getExtensionAddressFromTx(tx);
+      const extension = await TestExtension2.at(extensionAddress);
+
+      const identifier = await extension.identifier();
+      const version = await extension.version();
+      expect(identifier).to.equal(TEST_EXTENSION);
+      expect(version).to.eq.BN(2);
+    });
+
     it("does not allow an extension to be installed with a nonexistent resolver", async () => {
       await checkErrorRevert(colony.installExtension(TEST_EXTENSION, 0, { from: ROOT }), "colony-network-extension-bad-version");
     });
 
-    it("does not allow an extension to be installed twice", async () => {
-      await colony.installExtension(TEST_EXTENSION, 1, { from: ROOT });
+    it("allows colonies to migrate to multiExtension bookkeeping", async () => {
+      const extension = await TestExtension1.new();
+      await extension.install(colony.address);
+
+      let colonyAddress;
+
+      colonyAddress = await colonyNetwork.getExtensionMultiInstallation(extension.address);
+      expect(colonyAddress).to.equal(ethers.constants.AddressZero);
+
+      // Set up `installations` mapping in the old style
+      const slot = soliditySha3(`0x000000000000000000000000${colony.address.slice(2)}`, soliditySha3(TEST_EXTENSION, 39));
```

Comment:
`ethers.utils.zeroPad` could be our friend here (and elsewhere in these tests), I think!